scripts/cassandra-integration-test.sh

```diff
@@ -48,14 +52,16 @@ run_integration_test() {
   local major_version=${version%%.*}
   local schema_version=$2
   local jaegerVersion=$3
+  local username="username"
+  local password="password"
   local primaryKeyspace="jaeger_v1_dc1"
   local archiveKeyspace="jaeger_v1_dc1_archive"
   local compose_file="docker-compose/cassandra/v$major_version/docker-compose.yaml"
 
   setup_cassandra "${compose_file}"
```

Comment:
do the startup file for Cassandra does not need credentials to be defined, only apply-schema does?

Reply:
what do you mean by "startup file for Cassandra" ?
Is it this one in the docker.sh file  :  password and username.
![image](https://github.com/user-attachments/assets/d4bd9a94-8619-4735-a304-8f1a2216ee7a)

Would you please help.

Reply:
No, I was referring to `setup_cassandra "${compose_file}"` - this starts Cassandra via docker compose, but does not pass any credentials, so how does the server know later that username/password is what it's supposed to respect?

Reply:
I see,
Docker Compose has default configurations for Cassandra. It uses them for both development and test purposes, so I think we don't need to make changes in configurations as we can pass password/username in apply-schema()

Reply:
Local test done:
- scripts/cassandra-integration-test.sh 3 v003 v1
- scripts/cassandra-integration-test.sh 4 v004 v2

Reply:
> scripts/cassandra-integration-test.sh 3 v003 v1

this only proves my point. The v1 flavor of the test runs Go unit tests that internally initialize Cassandra connection and perform data writes / reads. You did not change those tests in your PR, which means they are creating Cassandra connection **without** a password. And yet your test still passes, so that does it prove?

What I expect is to run Cassandra in such a way that it will **reject** unauthenticated connections. Then the test will fail until you actually pass credentials from the code.

The v2 flavor of the test is different because it runs the actual binary with the config file, and you did change the config to provide credentials. But it doesn't provide any confidence that those credentials are actually used / respected / required - the only way to prove they are used is to remove them and demonstrate that the test fails without them.

Reply:
But this will require the CI also to pass credentials., will it ?

Reply:
yes, that's the point of this ticket.

Reply:
> What I expect is to run Cassandra in such a way that it will **reject** unauthenticated connections. Then the test will fail until you actually pass credentials from the code.


But Cassandra actually accept all connections even without credentials according to their [website](https://cassandra.apache.org/doc/stable/cassandra/operating/security.html#authentication)

Reply:
As default maybe, but it can be configured with auth.

Reply:
> As default maybe, but it can be configured with auth.

is there any way to do that, because environment variables does not work in the docker-compose.yaml 
and as I searched it[ is not supported ](https://github.com/docker-library/cassandra/issues/142#issuecomment-426772644)

Reply:
we may need to do it via custom Cassandra config